libs/animation/animation.ts

```diff
@@ -1,173 +1,624 @@
-
-            this.sprites.forEach(sprite => {
-                if (sprite._action === this.action) {
-                    let newImage = this.getImage();
-                    //Update only if the image has changed
-                    if (sprite.image !== newImage) {
-                        sprite.setImage(newImage);
+        public static parse(pathStart: Point, pathString: string): Path {
+            let path: Path = new Path();
+            let p0: Point = pathStart;
+
+            // This implementation of SVG parsing does not support the A/a commands, nor does it support exponents in arguments
+            const digits = "0123456789";
+            const separators = ", \t\n\r\f\v";
+            const commands: {
+                [ command: string ]: number
+            } = {
+                "M": 2, "m": 2, // moveTo
+                "L": 2, "l": 2, // lineTo
+                "H": 1, "h": 1, // horizontalLineTo
+                "V": 1, "v": 1, // verticalLineTo
+                "Q": 4, "q": 4, // quadraticCurveTo
+                "T": 2, "t": 2, // smoothQuadraticCurveTo
+                "C": 6, "c": 6, // cubicCurveTo
+                "S": 4, "s": 4, // smoothCubicCurveTo
+                // "A": 7, "a": 7, // arcTo
+                "Z": 0, "z": 0 // closePath
+            };
+            const signs = "+-";
+
+            let currentArg: string = "";
+            let command: string = null;
+            let args: number[] = [];
+
+            for (let i = 0; i < pathString.length; i++) {
+                const char = pathString.charAt(i);
+                const lastNode = path.nodes[path.nodes.length - 1];
+                
+                // This is an SVG path parser. It's kinda complicated. For each character, evaluate the following conditions:
+                // - if it's a digit, add it to the current argument
+                // - else if it's whitespace or newline, finish the current argument and prepare for the next one
+                // - else if it's a command, complete the previous argument, and prepare for the next one
+                //   - if there's sufficient data to make a node during this step, create it and continue
+                // - else if it's a plus/minus sign, and if it's the start of a new argument, add it to allow for positive/negative numbers
+                // - if it's the end of the string, complete the current argument before proceeding to the next step
+                // - if there's sufficient data to make a node after all of these steps, create it
+                if (digits.indexOf(char) > -1) { // Parses number arguments
+                    currentArg += char;
+                } else if (separators.indexOf(char) > -1 && currentArg) { // Terminates number arguments
+                    args.push(parseInt(currentArg));
+                    currentArg = "";
+                } else if (Object.keys(commands).indexOf(char) > -1) { // Parses command arguments
```

Comment:
This might change re: the last comment, but if not it seems preferable to pull out the ``Object.keys(commands)`` here into a const to avoid recreating the same array over and over again. If that last comment is implemented I would imagine you'd just return -1 in the default case there and handle based of that, though